src/structures/GuildEvent.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const Base = require('./Base');
+const GuildEventEntityMetadata = require('./GuildEventEntityMetadata');
+const { PrivacyLevels, GuildEventEntityTypes, GuildEventStatuses } = require('../util/Constants');
+
+/**
+ * Represents an event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildEvent extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {APIGuildEvent} data The data for the guild event
+   */
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The guild event's id
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    if ('channel_id' in data) {
+      /**
+       * The id of the stage channel this event belongs to
+       * @type {Snowflake}
+       */
+      this.channelId = data.channel_id;
+    }
+
+    /**
+     * The name of the guild event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild event
+       * @type {string}
+       */
+      this.description = data.description;
+    }
+
+    if ('image' in data) {
+      /**
+       * The image of the guild event
+       * @type {string}
+       */
+      this.image = data.image;
+    }
+
+    /**
+     * The time at which the guild event will start
+     * @type {Date}
+     */
+    this.scheduledStartTime = data.scheduled_start_time;
+
+    /**
+     * The time at which the guild event will end, or `null` if the event does not have a scheduled time to end
+     * @type {?Date}
+     */
+    this.scheduledEndTime = data.scheduled_end_time;
```

Comment:
Same reason as above, should also create a timestamp getter imo
```suggestion
    this.scheduledEndAt = new Date(data.scheduled_end_time);
```